### PR TITLE
DEP: remove deprecated loffset and base args for resample and Grouper

### PR DESCRIPTION
**Pull request by vamsi-verma-s**
- [x] resolves on of the issues listed in #48855
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x]  Added an entry in the latest doc/source/whatsnew/vX.X.X.rst file if fixing a bug or adding a new feature.

This is to enforce deprecation for `loffset` and `base` args which were deprecated in 1.1 (https://github.com/pandas-dev/pandas/pull/31809). 

removing loffset also resolves the pylint issue in #48855

---
Earlier Analysis:

There are two error raised by this check. Both in `Resampler._apply_loffset` for `self.loffset` . 

`loffset` is an attribute of `Grouper` . 

When `.resample` is called like below. A `Resampler` object is returned on a `Grouper` object rather than the `Series` (happens in get_resampler). Therefore, `loffset` attribute defined before access. 

```python
index = pd.date_range('1/1/2000', periods=9, freq='T')
series = pd.Series(range(9), index=index)
s = series.resample('3T')
```

https://github.com/pandas-dev/pandas/blob/75d91c850bd42ffbd8707065b7d3691b7a0cef99/pandas/core/resample.py#L1577-L1585

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
thanks for your PR

looks like loffset is deprecated anyway https://github.com/pandas-dev/pandas/pull/31809/files

Do you want to just enforce the deprecation, so there's no need to ignore the check? There's a bunch of examples of how to do this, e.g. https://github.com/pandas-dev/pandas/pull/49084

**Comment by vamsi-verma-s** [comment]
sure, it makes sense to enforce deprecation. i will go through the example and update.

**Comment by vamsi-verma-s** [comment]
Hi @MarcoGorelli, 

 looks like both `base` and `loffset` args were deprecated at the same time (https://github.com/pandas-dev/pandas/pull/31809)
should I go ahead and look into enforcing deprecation for `base` as well?

**Comment by MarcoGorelli** [comment]
yeah, sure

**Comment by vamsi-verma-s** [comment]
not sure why flake8 is failing. I am unable to reproduce it on local

https://github.com/pandas-dev/pandas/actions/runs/3284846670/jobs/5411280291#step:4:68

**Comment by vamsi-verma-s** [comment]
Hi @MarcoGorelli , can you please take a look. I have made the changes to remove both `loffset` and `base` args.
I am unable to understand why flake8 is failing.

**Comment by MarcoGorelli** [comment]
```
 flake8..................................................................................................Failed
- hook id: flake8
- exit code: 1

pandas/tests/resample/test_deprecated.py:2:1: F401 'pandas.date_range' imported but unused
```

looks like `date_range` is just unused in that file, you just need to remove it from the imports

**Comment by vamsi-verma-s** [comment]
> ```
>  flake8..................................................................................................Failed
> - hook id: flake8
> - exit code: 1
> 
> pandas/tests/resample/test_deprecated.py:2:1: F401 'pandas.date_range' imported but unused
> ```
> 
> looks like `date_range` is just unused in that file, you just need to remove it from the imports

it is used in one test here - 
https://github.com/pandas-dev/pandas/pull/49101/files#diff-30ff72cd26b5c624981dc932274a81b53484a5ecde656cd8741f76a4b614ef94R38

**Comment by vamsi-verma-s** [comment]
Thank you. I was looking at only the diff in this PR.

**Comment by vamsi-verma-s** [comment]
great. got to know this, my understanding was wrong on the diff- 

>This is a result of the type of diff we use on GitHub. We use git’s three dot diff, which is the difference between the latest commit on the HEAD branch and the last common ancestor commit with the base branch.

**Comment by vamsi-verma-s** [comment]
@MarcoGorelli Please review now.

**Comment by vamsi-verma-s** [comment]
Hi @MarcoGorelli , could you please review this.

File: doc/source/whatsnew/v2.0.0.rst [hunk 1]
```diff
@@ -292,6 +292,7 @@ Groupby/resample/rolling
 - Bug in :meth:`DataFrameGroupBy.sample` raises ``ValueError`` when the object is empty (:issue:`48459`)
 - Bug in :meth:`Series.groupby` raises ``ValueError`` when an entry of the index is equal to the name of the index (:issue:`48567`)
 - Bug in :meth:`DataFrameGroupBy.resample` produces inconsistent results when passing empty DataFrame (:issue:`47705`)
+- Removed the previously deprecated ``base`` and ``loffset`` arguments from :meth:`pandas.DataFrame.resample`, :meth:`pandas.Series.resample` and :class:`pandas.Grouper`. Use ``offset`` or ``origin`` instead (:issue:`31809`)
```
Comment (MarcoGorelli) [inline doc/source/whatsnew/v2.0.0.rst:295]: there's a `Removal of prior version deprecations/changes` section

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
generally looks good (sorry for the delay, was away)

File: doc/source/whatsnew/v2.0.0.rst (hunk 1 above)
Comment (vamsi-verma-s) [reply doc/source/whatsnew/v2.0.0.rst]: Thank you. I have moved the change note to deprecations section.

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me, thanks @vamsi-verma-s 



Leaving open for a bit in case others have comments

**Pull request merged**